airbyte-integrations/connectors/source-hubspot-singer/README.md

```diff
@@ -0,0 +1,61 @@
+# Hubspot Test Configuration
+
+This integration wraps the existing singer [tap-hubspot](https://github.com/singer-io/tap-hubspot). In order to test the Hubspot source, you will need api credentials, see the [docs](https://docs.airbyte.io/integrations/sources/hubspot) for details.
+
+## Community Contributor
+
+1. Create a file at `secrets/config.json` with one of the following two format:
+    1. If using an api key:
+        ```
+        {
+          "start_date": "2017-01-01T00:00:00Z",
+          "credentials": {
+            "api_key": "your hubspot api key"
+          }
+        }
+        ```
+   1. If using oauth:
+       ```
+        {
+          "start_date": "2017-01-01T00:00:00Z",
+          "credentials": {
+            "redirect_uri": "<redirect uri>",
+            "client_id": "<hubspot client id>",
+            "client_secret": "<hubspot client secret>",
+            "refresh_token": "<hubspot refresh token>",
+          }
+        }
+       ```
+
+## Airbyte Employee
+
+1. Access the api key credentials in the `hubspot-integration-test-api-key` secret on Rippling under the `Engineering` folder
+    1. If we ever need a new api key it can be found in settings -> integrations (under the account banner) -> api key
+1. Access the oauth config in the `hubspot-integration-test-oauth-config` secret on Rippling under the `Engineering` folder
+    1. If we ever need to regenerate the refresh token for auth follow these hubspot [instructions](https://developers.hubspot.com/docs/api/oauth-quickstart-guide). Going to lay out the process because it wasn't 100% clear in their docs.
```

Comment:
Isn't the refresh token something that need to be generated from the integration itself instead of an input?

Reply:
Ah, as in could we do the oauth within our UI and generate the refresh_token ourselves? Good thought. 2 things:

1. It is not uncommon for singer taps / targets to require a refresh token. That's because there's no UI for it to do a normal oauth flow so it cheeses the oauth flow. It seems fairly standard in singer world. (both the tap-salesforce and tap-hubspot use this pattern).
1. Since we have the UI, we don't necessarily need to be bound by this constraint, but we would have to add a new feature to the UI to support doing oauth and add this to the integration interface. I think I'd be in favor of doing this later, since:
    1. it's always already common to do it the current way in these tools and 
    1. will be a non trivial change in both the ui and integration iface 
    1. it will only make it easier to use our integrations, we'll just remove refresh_token from configs, so it's an easy migration for users.

does this address what you were thinking of? or am i misunderstanding?

Reply:
It does address my question.

Reply:
Created an issue to track us supporting oauth in a more use-friendly way: https://github.com/airbytehq/airbyte/issues/768